CONTRIBUTING.md

```diff
@@ -0,0 +1,207 @@
+- Address the issue
+- Commit changes
+- Open a pull request
+- Have a peer review the pull request
+    - A pull request may require revisions from the reviewer before being approved 
+- Merge approved pull request into the main branch
+- Delete the working branch if no longer needed
+
+## Coding Standards and Style Guides
+
+### General Principles (All Languages)
+- Write clear, readable, and self-explanatory code.
+- Use consistent indentation.
+- Use meaningful names for variables, functions, and classes.
+- Avoid hard-coding values - use constants or configuration where appropriate.
+- Comment why something is done, not what is done (code should explain the "what").
+- Avoid global variables unless necessary.
+
+### Fortran
+
+#### File Naming
+- Use ```.f90``` for modern Fortran source files.
+- Lowercase file names with underscores: ```calculate_area.f90```
+
+#### Style
+- Use ```implicit none``` to avoid undeclared variables.
+- Group related code into modules and subroutines.
+ 
+#### Naming
+- Use ```snake_case``` for variables and subroutines.
+- Modules: ```module_name```
+- Constants:```ALL_CAPS```
+
+#### Example
+```
+module circle_utils
+  implicit none
+  real, parameter :: PI = 3.14
+contains
+  function calculate_area(radius) result(area)
+    real, intent(in) :: radius
+    real :: area
+    area = PI * radius * radius
+  end function calculate_area
+end module circle_utils
+```
+
+### Java
+
+#### File Naming
+- Each class should go in its own ```.java``` file named after the class.
+
+#### Style
+- Follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+
+#### Naming
+- Classes: ```PascalCase```
+- Methods and variables: ```camelCase```
+- Constants: ```UPPER_SNAKE_CASE``` (with ```static final```)
+
+#### Example
+```
+public class Circle {
+    public static final double PI = 3.14;
+
+    public double calculateArea(double radius) {
+        return PI * radius * radius;
+    }
+}
+```
+### Python
+
+#### File Naming
+- Use ```snake_case.py``` for file names.
+
+#### Style
+- Follow guidelines in the [PEP 8](https://peps.python.org/pep-0008/).
+- Use 4-space indentation.
+- Limit lines to 79 characters.
+
+#### Naming
+- Variables and functions: ```snake_case```
+- Classes: ```PascalCase```
+- Constants: ```UPPER_SNAKE_CASE```
+
+#### Example
+```
+def calculate_area(radius):
+    """Calculate area of a circle."""
+    pi = 3.14
+    return pi*radius**2
+```
+
+## Branching Strategies
+
+### ```main``` Branch
+- Keep ```main``` always **stable and production-ready**.
+- Support **collaborative development**.
+- Ensure features and fixes are tested before release.
+- Require PR reviews.
+- Disallow direct pushes.
+
+## Git Operations and Pull Requests Guidelines
+
+### Branching
+
+- All branches should have a known goal and lifecycle.
+- The branch name should summarize its goal and be appended by:
+    - devops/ - branches with changes that do not actually affect source code
```

Comment:
needs an explanation of what kind of change this would actually be (instead of just what is isn't)